File: .vscode-test/cdk-template.py
from aws_cdk import (
    Stack,
)
from aws_cdk import (
    aws_iam as iam,
    aws_kms as kms,
)
from constructs import Construct


class MyStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        sns_key = kms.Key(self, "snsTopicKey",
            enable_key_rotation=True,
            alias="snsTopicKey",
            policy=iam.PolicyDocument(
                statements=[
                    iam.PolicyStatement(
                        actions=["kms:Decrypt", "kms:GenerateDataKey*", "kms:Encrypt", "kms:ReEncrypt*", "kms:DescribeKey"],
                        resources=["*"],
                        effect=iam.Effect.ALLOW,
                        principals=[
                            iam.ServicePrincipal("cloudwatch.amazonaws.com"),
                            iam.ServicePrincipal("sns.amazonaws.com"),
                            iam.ServicePrincipal(""),
                        ]
                    ),
                    iam.PolicyStatement(
                        actions=["kms:*"],
                        resources=["*"],
                        effect=iam.Effect.ALLOW,
                        principals=[iam.AccountRootPrincipal()]
                    )
                ]
            )
        )

        test_role = iam.Role(self, "testRole", assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"))

        test_policy = iam.PolicyStatement(
            actions=[
                "sagemaker:DeleteApp",
                "sagemaker:DeleteSpace",
            ],
            resources=["*"],
            effect=iam.Effect.ALLOW,
            principal=iam.ServicePrincipal("a2c.amazonaws.com"),
        )
        test_role.add_to_principal_policy(test_policy)

File: .vscode-test/cdk-template.ts
import * as cdk from 'aws-cdk-lib';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as kms from 'aws-cdk-lib/aws-kms';
import { Construct } from 'constructs';

export class TestStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const testRole = new iam.Role(this, 'testRole', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
    });

    const snsKey = new kms.Key(this, 'snsTopicKey', {
      enableKeyRotation: true,
      alias: 'snsTopicKey',
      policy: new iam.PolicyDocument({
        statements: [
          new iam.PolicyStatement({
            actions: ['kms:Decrypt', 'kms:GenerateDataKey*', 'kms:Encrypt', 'kms:ReEncrypt*', 'kms:DescribeKey'],
            resources: ['*'],
            effect: iam.Effect.ALLOW,
            principals: [
              new iam.ServicePrincipal('cloudwatch.amazonaws.com'),
              new iam.ServicePrincipal('sns.amazonaws.com'),
              new iam.ServicePrincipal(''),
            ],
          }),
          new iam.PolicyStatement({
            actions: ['kms:*'],
            resources: ['*'],
            effect: iam.Effect.ALLOW,
            principals: [new iam.AccountRootPrincipal()],
          }),
        ],
      }),
    });

    const testPolicy = new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      actions: [
        'sagemaker:DeleteApp',
        'sagemaker:DeleteSpace',
        'sagemaker:DescribeApp',
        'sagemaker:DescribeSpace',
        'sagemaker:CreateEndpoint',
      ],
      resources: ['*'],
      principal: new iam.Principal('access-analyzer.amazonaws.com'),
    });
    testRole.addToPolicy(testPolicy);
  }
}
